### PR TITLE
zscore code revision

### PR DESCRIPTION
**Pull request by lvonasek**
By the code revision I found those issues:

* FATAL comparing rounded and unrounded height (rule == lastMeasure.getHeight()))
* FATAL comparing rounded and unrounded age (rule == lastMeasure.getAge() / 30)
* height ZScore for (Zind < -3 || Zind > 3) was calculated incorrectly
* age was calculated inaccurately (1 month difference for 6 year old)
* bad practice (boys = boys_0_2) could lead that on second run only boys_0_2 are used for boys

### CODE REVIEW

File: app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java [hunk 1]
```diff
@@ -463,24 +465,25 @@ public void setData() {
...
-        }
-        else if(zscore>3){
-            finalZscore=3+((value-SD3pos)/SD23pos);
-        }
-        else{
-            finalZscore=-3+((value-SD3neg)/SD23neg);
-        }
-        return finalZscore;
+    //Compute ZScore as defined in https://www.who.int/growthref/computation.pdf
+    private double heightZScore(double y, double Mt, double Lt, double St) {
```
Comment (prajurock) [inline app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java:469]: image
Why no comparison as shown in the image?

**Review by prajurock — COMMENTED** [review]
Thank you for this PR. Major fault.

File: app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java (hunk 1 above)
Comment (lvonasek) [reply app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java:469]: If you read the PDF carefully, then you see that:
* the image you posted shows "z-scores for weight-for-age and BMI-for-age"
* for height we should use the one for "z-scores for height-for-age" (page 1)

Comment (prajurock) [reply app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java:469]: Yes. I checked the way it is written is quite confusing with the title but I assured you it has to be there.

Comment (lvonasek) [reply app/src/main/java/de/welthungerhilfe/cgm/scanner/ui/fragments/GrowthDataFragment.java]: okay, I put it there.

**Review by prajurock — APPROVED** [review]

**Pull request merged**